master/internal/rm/agent_resource_manager.go

```diff
@@ -71,6 +71,23 @@ func (a AgentResourceManager) ValidateResourcePool(ctx actor.Messenger, name str
 	return err
 }
 
+// IsReattachEnabled is true for the AgentResourceManager.
+func (a AgentResourceManager) IsReattachEnabled(ctx actor.Messenger) bool {
+	return true
+}
```

Comment:
```suggestion
func (a AgentResourceManager) IsReattachEnabled(ctx actor.Messenger) bool {
	config := config.GetMasterConfig()

	for _, rpConfig := range config.ResourcePools {
		if rpConfig.AgentReattachEnabled {
			return true
		}
	}
	return false
}
```

Reply:
> I don't understand why the concept of agent being able to potentially Reattach necessarily has to align with if any resource_pool is configured to do so. 

It is 100% related, an agent is potentially able to reattach if and only if a resource pool supports it. I will refactor this eventually, I suppose.

> The place where IsReattachEnabled is called, is part of trial/restore and immediately followed by IsReattachEnabledForRP for the give partition, so we are just duplicating the check

seems like trial should just call `IsReattachEnabledForRP` then